R/btw2021.R

```diff
@@ -0,0 +1,331 @@
+  file = "data/btw2021/erststimmenNachPartei.csv",
+  col_types = cols(
+    stimmbezirk = readr::col_factor(),
+    stimmbezirkNr = readr::col_factor(),
+    partei = readr::col_factor(levels = levels(parteien$Kuerzel)),
+    stimmen = col_integer()
+  )
+) %>%
+  inner_join(parteien, by = join_by(partei == Kuerzel)) %>%
+  inner_join(direktkandidaten %>% transmute(
+    Partei,
+    Direktkandidat = Name
+  ), by = join_by(partei == Partei)) %>%
+  inner_join(erststimmenAllgemein %>% select(stimmbezirk, gueltigeStimmen), by = "stimmbezirk") %>%
+  mutate(stimmenAnteil = stimmen/gueltigeStimmen)
+
+
+zweitstimmenAllgemein <- read_csv(
+  file = "data/btw2021/zweitstimmenAllgemein.csv",
+  col_types = cols(
+    stimmbezirk = readr::col_factor(),
+    stimmbezirkNr = readr::col_factor(),
+    stimmbezirkArt = readr::col_factor(),
+    wahlberechtigte = col_integer(),
+    waehler = col_integer(),
+    ungueltigeStimmen = col_integer(),
+    gueltigeStimmen = col_integer()
+  )
+)
+
+zweitstimmenNachStimmbezirkArt <- zweitstimmenAllgemein %>%
+  filter(!is.na(stimmbezirkArt)) %>%
+  group_by(stimmbezirkArt) %>%
+  summarise(
+    waehler = sum(waehler),
+    ungueltigeStimmen = sum(ungueltigeStimmen),
+    gueltigeStimmen = sum(gueltigeStimmen)
+  )
+
+zweitstimmenNachPartei <- read_csv(
+  file = "data/btw2021/zweitstimmenNachPartei.csv",
+  col_types = cols(
+    stimmbezirk = readr::col_factor(),
+    stimmbezirkNr = readr::col_factor(),
+    partei = readr::col_factor(levels = levels(parteien$Kuerzel)),
+    stimmen = col_integer()
+  )
+) %>%
+  inner_join(parteien, by = join_by(partei == Kuerzel)) %>%
+  inner_join(zweitstimmenAllgemein %>% select(stimmbezirk, gueltigeStimmen), by = "stimmbezirk") %>%
+  mutate(stimmenAnteil = stimmen/gueltigeStimmen)
+
+
+
+ui <- memoise(omit_args = "request", function(request, id) {
+  request <- NULL # unused variable, so we set it to NULL to avoid unintended usage
+
+  ns <- NS(id)
+  tagList(
+    h2("Bundestagswahl 26. September 2021 in der Gemeinde Vaterstetten"),
+
+    fluidRow(
+      box(
+        title = "Hinweise zu den Stimmbezirken",
+        status = "primary",
+        solidHeader = TRUE,
+        width = 12,
+        tagList(
+          p(HTML("Eingezeichnet sind die 14 Wahllokal-Stimmbezirke, die auch in <a href=\"https://umap.openstreetmap.fr/de/map/bundestagswahl-2021-stimmbezirke_659020#14/48.1192/11.7997\">dieser (inoffiziellen) Karte</a> im Detail angesehen werden können. Nicht eingezeichnet sind die Briefwahlbezirke (31-43). Da das Wahlverhalten der Briefwähler (ca 2/3 der Wähler!) nicht dargestellt werden kann, führt dies möglicherweise zu Verzerrungen in der Darstellung. Weiter ist zu beachten, dass manche Stimmbezirke teils sehr kleine isolierte Gebiete aufweisen (wie der Hofkellermeisterweg oder das Gut Ammerthal) – das dortige Ergebnis entspricht dennoch dem Durchschnitt im ganzen Stimmbezirk, es ist also keine punktuelle Interpretation möglich.")),
```

Comment:
```suggestion
          p(HTML("Eingezeichnet sind die 14 Wahllokal-Stimmbezirke, die auch in <a href=\"https://umap.openstreetmap.fr/de/map/bundestagswahl-2021-stimmbezirke_659020#14/48.1192/11.7997\">dieser (inoffiziellen) Karte</a> im Detail angesehen werden können. Nicht eingezeichnet sind die Briefwahlbezirke (31-43). Da das Wahlverhalten der Briefwähler (ca ⅔ der Wähler!) nicht dargestellt werden kann, führt dies möglicherweise zu Verzerrungen in der Darstellung. Weiter ist zu beachten, dass manche Stimmbezirke teils sehr kleine isolierte Gebiete aufweisen (wie der Hofkellermeisterweg oder das Gut Ammerthal) – das dortige Ergebnis entspricht dennoch dem Durchschnitt im ganzen Stimmbezirk, es ist also keine punktuelle Interpretation möglich.")),
```

Der letzte Satz ist außerdem etwas kompliziert formuliert.